### PR TITLE
Interpolation modes documentation upgrade

### PR DESCRIPTION
**Pull request by szalpal**
## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
This PR upgrades documentation of the interpolation modes. Please refer to the changed files for more details.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [x] Doxygen
  - [ ] RST
  - [x] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

File: dali/python/backend_impl.cc [hunk 1]
```diff
@@ -1792,7 +1792,10 @@ PYBIND11_MODULE(backend_impl, m) {
     .export_values();
 
   // DALIInterpType
-  py::enum_<DALIInterpType>(types_m, "DALIInterpType", "Interpolation mode\n<SPHINX_IGNORE>")
+  py::enum_<DALIInterpType>(types_m, "DALIInterpType", "Interpolation mode.\n"
+          "Linear and Cubic interpolations in DALI are separable with regards to dimensions. "
```
Comment (stiepan) [inline dali/python/backend_impl.cc:1796]: Nitpick: All of them are separable, right? I understand that the two names are used with bi- tri- regularly, but maybe in the first sentence we could just say "The interpolation modes are separable with regards to dimensions"? Not to make the impression the two are treated specially.

Comment (mzient) [reply dali/python/backend_impl.cc:1796]: Any person with enough knowledge to even attempt to run DALI will know that "linear" is bilinear for images. Mentioning "trilinear" will actually cause confusion for anyone with 3D graphics background, where trilinear refers to linearly interpolating two nearest bi-linearly-interpolated mip-maps.
Also, whether or not we use a separable implementation is operator-dependent. Resize uses separable convolutions, but warp does it in one go.

Comment (szalpal) [reply dali/python/backend_impl.cc:1796]: @mzient ,

Thanks for thorough explanation. I believe that my proposal on this documentation is sufficient, however if you have any other suggestion I'll be happy to apply it. Would you mind sharing the suggestion or may we proceed with current version?

Comment (mzient) [reply dali/python/backend_impl.cc:1796]: OK. I would refrain from elaborating on what we do in the general description and add per-value docstrings instead.
```Python
    .value("INTERP_NN", DALI_INTERP_NN, "Nearest neighbor")
    .value("INTERP_LINEAR", DALI_INTERP_LINEAR, "Linear interpolation. Synonymous with bilinear for 2D inputs.")
    .value("INTERP_CUBIC", DALI_INTERP_CUBIC, "Cubic interpolation. Synonymous with bicubic for 2D inputs.")
    .value("INTERP_LANCZOS3", DALI_INTERP_LANCZOS3, "Resampling with a Lanczos window with 3 lobes.")
    .value("INTERP_TRIANGULAR", DALI_INTERP_TRIANGULAR, "Resampling with a triangular window.")
    .value("INTERP_GAUSSIAN", DALI_INTERP_GAUSSIAN, "Resampling with a Gaussian window.")
```

File: dali/python/backend_impl.cc [hunk 2]
```diff
@@ -1792,7 +1792,10 @@ PYBIND11_MODULE(backend_impl, m) {
...
+          "Therefore, ``DALI_INTERP_LINEAR`` and ``DALI_INTERP_CUBIC`` modes capture all: "
+          "Linear, Bi-linear, Tri-linear etc... interpolation modes.\n<SPHINX_IGNORE>")
```
Comment (mzient) [inline dali/python/backend_impl.cc:1798]: If Sphinx can't read per-value docstrings, I'd simplify the message:
```suggestion
  py::enum_<DALIInterpType>(types_m, "DALIInterpType", "Interpolation mode.\n"
          "Note: for 2D inputs, linear and cubic are synonymous with bilinear and bicubic, respectively.")
```

File: dali/python/backend_impl.cc (hunk 1 above)
Comment (szalpal) [reply dali/python/backend_impl.cc]: Added those

File: dali/python/backend_impl.cc (hunk 2 above)
Comment (szalpal) [reply dali/python/backend_impl.cc]: Changed this too

**Comment by szalpal** [comment]
!build

**Review by mzient — APPROVED** [review]

**Review by stiepan — APPROVED** [review]

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

**Pull request merged**